packages/es-dev-server/src/utils/resolve-module-imports.js

```diff
@@ -184,18 +184,17 @@ async function resolveModuleImportsWithConfig(importer, source, cfg) {
       lastIndex = end;
     } else if (dynamicImportIndex >= 0) {
       // dynamic import
-      const dynamicStart = start + 1;
-      const dynamicEnd = end - 1;
-
+      const literals = [`\``, "'", '"'];
+      const stringSymbol = literals.find(char => source.substring(start, end).includes(char));
```

Comment:
`source.substring(start, end)` should be extracted to a separate var, otherwise we're executing it on each iteration. You can then perform all the operations below on this substring, since it's a smaller subset it will be faster.

Reply:
The find operation should check the first character which matches any of the literals, instead of checking the first literal that matches any of the characters in the string.

Reply:
The result of this can be `dynamicStart`, then you can derive the `stringSymbol` from that. That saves a bit of work :)

```js
const dynamicStart = .. whatever magic ...
const stringSymbol = source[dynamicStart];
```

We will however need handle the case when `dynamicStart === -1` because then there is no string literal in the import. In that case, we should skip a lot of the work below.